src/output.rs

```diff
@@ -51,29 +45,41 @@ fn print_entry_colorized(
     path: &Path,
     config: &Options,
     ls_colors: &LsColors,
-    wants_to_quit: &Arc<AtomicBool>,
 ) -> io::Result<()> {
-    let default_style = ansi_term::Style::default();
-
-    // Traverse the path and colorize each component
-    for (component, style) in ls_colors.style_for_path_components(path) {
-        let style = style
-            .map(Style::to_ansi_term_style)
-            .unwrap_or(default_style);
+    // Split the path between the parent and the last component
+    let mut offset = 0;
+    let path_str = path.to_string_lossy();
+
+    if let Some(parent) = path.parent() {
+        offset = parent.to_string_lossy().len();
+        for c in path_str[offset..].chars() {
+            if std::path::is_separator(c) {
+                offset += c.len_utf8();
+            } else {
+                break;
+            }
+        }
+    }
 
-        let mut path_string = component.to_string_lossy();
+    if offset > 0 {
+        let mut parent_str = Cow::from(&path_str[..offset]);
         if let Some(ref separator) = config.path_separator {
-            *path_string.to_mut() = replace_path_separator(&path_string, separator);
+            *parent_str.to_mut() = replace_path_separator(&parent_str, separator);
         }
-        write!(stdout, "{}", style.paint(path_string))?;
 
-        // TODO: can we move this out of the if-statement? Why do we call it that often?
-        if wants_to_quit.load(Ordering::Relaxed) {
```

Comment:
I think I have misinterpreted the reason for this code and we need to keep this check